controllers/packetcluster_controller.go

```diff
@@ -97,30 +97,44 @@ func (r *PacketClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 		}
 	}()
 
-	clusterScope.PacketCluster.Status.Ready = true
+	// Handle deleted clusters
+	if !cluster.DeletionTimestamp.IsZero() {
+		return r.reconcileDelete(ctx, logger, clusterScope)
+	}
+
+	return r.reconcile(ctx, logger, clusterScope)
+}
 
+func (r *PacketClusterReconciler) reconcile(ctx context.Context, logger logr.Logger, clusterScope *scope.ClusterScope) (ctrl.Result, error) {
 	address, err := r.getIP(clusterScope.PacketCluster)
+	if err == nil {
+		clusterScope.PacketCluster.Status.APIEndpoints = []infrastructurev1alpha3.APIEndpoint{
+			{
+				Host: address,
+				Port: 6443,
+			},
+		}
+		clusterScope.PacketCluster.Status.Ready = true
```

Comment:
Move this up. It is ok (desired) to have the cluster ready _before_ we have the control plane node up. This is precisely why we wait with the workers. They cannot do much without it. So this part should be reverted. `Ready` as soon as we start, APIEndpoint when we have it.